cli/flox-rust-sdk/src/models/lockfile.rs

```diff
@@ -81,6 +81,24 @@ pub struct Registry {
     _json: Value,
 }
 
+#[derive(Debug, Clone)]
+pub enum LockResult {
+    /// Locking produced a new Lockfile.
+    /// The change could be a minimal as whitespace.
+    Changed(Lockfile),
+    /// Locking did not produce a new Lockfile.
+    Unchanged(Lockfile),
+}
+
+impl From<LockResult> for Lockfile {
+    fn from(result: LockResult) -> Self {
+        match result {
+            LockResult::Changed(lockfile) => lockfile,
+            LockResult::Unchanged(lockfile) => lockfile,
+        }
+    }
+}
```

Comment:
(nb)

I'd also debate whether to use `Into` vs accessors `{to,into}_lockfile`.
For now i think `Into` is doing the job.
An AsRef impl might be useful to get .. a reference instead.